.github/workflows/udmp-parser.yml

```diff
@@ -2,74 +2,117 @@ name: Builds
 
 on: [push, pull_request]
 
+
 jobs:
   Windows:
-    runs-on: windows-latest 
+    runs-on: windows-latest
     strategy:
       fail-fast: false
       matrix:
         generator: ['ninja', 'msvc']
-        arch: ['win32', 'win64']
+        arch: ['x64'] #, 'x86', 'arm64']
+        python-version: [3.9]
         exclude:
           # Ninja doesn't support the -A arch options with CMake
           - generator: ninja
-            arch: win32
-
+            arch: x86
+    env:
+      CMAKE_CONFIG: RelWithDebInfo
     name: Windows latest / ${{ matrix.generator }}.${{ matrix.arch }}
     steps:
     - name: Checkout
       uses: actions/checkout@v3
 
-    - name: Add msbuild to PATH
-      uses: microsoft/setup-msbuild@v1
+    - name: Setup Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
 
-    - name: Setup vs prompt
-      uses: ilammy/msvc-dev-cmd@v1
+    - name: Install Python pre-requisites
+      shell: pwsh
+      run: |
+        python -m pip install --upgrade pip setuptools wheel
+        python -m pip install --user --upgrade -r src/python/requirements.txt
 
     - name: Build with Ninja/cl
       if: matrix.generator == 'ninja'
       run: |
         cd src\build
         .\build-release.bat
 
-    - name: Build with msvc
+    - name: Build (msvc)
       if: matrix.generator == 'msvc'
+      shell: pwsh
       run: |
-        cd src\build
-        .\build-release-msvc.bat ${{ matrix.arch }}
+        Import-Module .\.github\Invoke-VisualStudio.ps1
+        Invoke-VisualStudio2022${{ matrix.arch }}
+        mkdir ./build
+        cmake -S ./src -B ./build -A ${{ matrix.arch }} ${{ env.CMAKE_CONFIG }}
+        cmake --build ./build --verbose --config ${{ env.CMAKE_CONFIG }}
+        cmake --install ./build --config ${{ env.CMAKE_CONFIG }} --prefix ./artifact --verbose
+
+    - name: Python Binding Tests (msvc)
+      if: matrix.generator == 'msvc'
+      shell: pwsh
```

Comment:
You're right, powershell is the default. It's just a habit I took because I've been bitten by shit like that in the past (and also, explicit is better than implicit 😄 ) Regardless, I'll remove

Reply:
I didn't know you were a powershell nerd hugsy :D